SIS_import/sis_set_url.rb

```diff
@@ -268,8 +269,20 @@ def ESB_APICall(url, authorization_string, content_type, request_type, param_has
 	# increase the call count by one
 	@esb_call_hash['call_count'] = @esb_call_hash['call_count'] +1
 
-	# return json
-	return json_parse_safe(url, response.body, @logger)
+	return response.body
+end
+
+## checks the result of ESB API call
+## renew ESB token if necessary, and retry the failed call due to expired token
+## return parsed JSON
+def parse_ESB_API_CALL_RESPONSE(responseBody, url, content_type, request_type, param_hash)
+	if (responseBody.include? "Invalid Credentials")
```

Comment:
Check response data for 401 (unauthorized) status instead of looking for text that might change.